pkg/promlib/resource/resource_test.go

```diff
@@ -107,3 +113,77 @@ func TestResource_GetSuggestions(t *testing.T) {
 	require.NoError(t, err)
 	assert.NotNil(t, resp)
 }
+
+func TestResource_GetSuggestionsWithEmptyQueriesButFilters(t *testing.T) {
```

Comment:
Extracting the code responsible for constructing the request URL, would make this test a bit simpler. I could do a followup PR, as I am guessing it will need a round of feedback or two. Let's get this PR merged ASAP, as it is blocking.